src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -0,0 +1,506 @@
+
+        self._s3_folder = s3_destination_folder
+        self._poll_timeout_seconds = poll_timeout_seconds
+        self._poll_interval_seconds = poll_interval_seconds
+
+
+    @property
+    def hardware_capabilities(self):
+        """Dictionary of hardware capabilities for the hardware device"""
+        return self._device.properties.paradigm.dict()
+
+    @property
+    def settings(self):
+        """Dictionary of constants set by the hardware.
+
+        Used to enable initializing hardware-consistent Hamiltonians by saving
+        all the values that would need to be passed, i.e. :
+
+            >>> dev_remote = qml.device('braket.aws.ahs', wires=3)
+            >>> dev_pl = qml.device('default.qubit', wires=3)
+            >>> settings = dev_remote.settings
+            >>> H_int = rydberg_interaction(coordinates, **settings)
+
+        If `H_int` is included in a hamiltonian used on the PennyLane `default.qubit` device,
+        it will use the harware-specific constants.
+        """
+        return {"interaction_coeff": self._get_rydberg_c6()}
+
+    def _get_rydberg_c6(self):
+        """Get rydberg C6 and convert from rad/s m^6 (AWS units) to MHz um^6 (PL simulation units)"""
+        c6 = float(self.hardware_capabilities['rydberg']['c6Coefficient'])  # rad/s x m^6
+        c6 = 1e-6 * c6/(2*np.pi)  # rad/s --> MHz
+        c6 = c6 * 1e36  # m^6 --> um^6
+        return c6
+
+    def create_ahs_program(self, evolution):
+        """Create AHS program for upload to hardware from a ParametrizedEvolution
+
+        Args:
+            evolution (ParametrizedEvolution): the PennyLane operator describing the pulse
+                to be converted into an Analogue Hamiltonian Simulation program
+
+        Returns:
+            AnalogHamiltonianSimulation: a program containing the register and drive
+                information for running an AHS task on simulation or hardware"""
+
+        ahs_program = self._ahs_program_from_evolution(evolution)
+        ahs_program_discretized = ahs_program.discretize(self._device)
+
+        self.ahs_program = ahs_program_discretized
+
+        return ahs_program_discretized
+
+    def _run_task(self, ahs_program):
+        task = self._device.run(
+            ahs_program,
+            s3_destination_folder=self._s3_folder,
+            shots=self.shots,
+            poll_timeout_seconds=self._poll_timeout_seconds,
+            poll_interval_seconds=self._poll_interval_seconds,
+        )
+        return task
+
+
+class BraketLocalAhsDevice(BraketAhsDevice):
+    """Amazon Braket LocalSimulator AHS device for PennyLane.
+
+    Args:
+        wires (int or Iterable[Number, str]]): Number of subsystems represented by the device,
+            or iterable that contains unique labels for the subsystems as numbers
+            (i.e., ``[-1, 0, 2]``) or strings (``['ancilla', 'q1', 'q2']``).
+        shots (int): Number of executions to run to aquire measurements. Defaults to 100.
```

Comment:
To confirm, the simulator cannot have `shots=None`?

Reply:
Correct, shots has to be an integer for the simulator